judges/pipeline.py

```diff
@@ -1,13 +1,199 @@
 """Python script responsible for extracting judge data using web scraping"""
 
 from os import environ as ENV
-from time import sleep
 import logging
+from datetime import datetime
 from dotenv import load_dotenv
 import requests
 from bs4 import BeautifulSoup
+import pandas as pd
+from psycopg2 import connect
+from psycopg2.extras import RealDictCursor
 
 
+# ========== FUNCTIONS: SCRAPING ==========
+def scrape_HCKB(url: str) -> pd.DataFrame:
+    """Get data from a list of judges with a URL"""
+
+    try:
+        response = requests.get(url, timeout=10)
+
+        soup = BeautifulSoup(response.content, 'html.parser')
+
+        cells = soup.find_all(
+            "td", class_="govuk-table__cell")
+        cells = [cell.text
+                 for cell in cells
+                 if "<strong>" not in str(cell)]
+
+        rows = []
+        for c in range(0, len(cells), 2):
+            rows.append({"judge": cells[c],
+                         "appointment": cells[c+1]})
+
+        return pd.DataFrame(rows)
+
+    except requests.RequestException as error:
+        logging.info(f"Error fetching URL: {error}")
+        return pd.DataFrame([])
+
+
+def scrape_CJ(url: str) -> pd.DataFrame:
+    """Get data from a list of judges with a URL"""
+
+    try:
+        response = requests.get(url, timeout=10)
+
+        soup = BeautifulSoup(response.content, 'html.parser')
+
+        cells = soup.find_all(
+            "td", class_="govuk-table__cell")
+        cells = [cell.text
+                 for cell in cells
+                 if "strong" not in str(cell)]
+
+        rows = []
+        for c in range(0, len(cells), 3):
+            rows.append({"judge": cells[c],
+                         "circuit": cells[c+1],
+                         "appointment": cells[c+2]})
+
+        return pd.DataFrame(rows)
+
+    except requests.RequestException as error:
+        logging.info(f"Error fetching URL: {error}")
+        return pd.DataFrame([])
+
+
+# ========== FUNCTIONS: TRANSFORMING ==========
+def convert_date(datestr: str) -> datetime:
+    """Turns date string to datetime.
+    Returns date as datetime."""
+
+    datestr = datestr.split()[0]
+    length = len(datestr)
+
+    if length == 10:
+        return datetime.strptime(datestr, "%d-%m-%Y")
+    if length == 8:
+        return datetime.strptime(datestr, "%d-%m-%y")
+    if length == 9:
+        return datetime.strptime(datestr, "%d-%b-%y")
```

Comment:
Why are there varying numbers when comparing the length of the datetime?

Reply:
because there's sadly a mix of formats on the government website